app/forms/steps/client/has_partner_form.rb

```diff
@@ -11,10 +11,30 @@ def choices
 
       private
 
+      # TODO: When FeatureFlag.partner_journey is removed,
+      # move this form into /partner and Partner module
       def persist!
-        crime_application.update(
-          attributes
-        )
+        ::CrimeApplication.transaction do
+          reset!
+
+          crime_application.update!(
+            attributes
+          )
+
+          true
+        end
+      end
+
+      def reset!
+        if client_has_partner.no?
+          crime_application.partner&.destroy!
+          crime_application.partner_detail&.destroy!
+        elsif client_has_partner.yes?
+          crime_application.partner_detail&.update!(
+            'relationship_status' => nil,
+            'separation_date' => nil
+          )
+        end
```

Comment:
I think this will run every time the form is submitted even if the answer hasn't changed. Is it worth checking if the answer has changed first?